### PR TITLE
Containers: enable testing for SLE Micro Rancher image

### PR DESCRIPTION
**Pull request by jlausuch**
- Related ticket: https://progress.opensuse.org/issues/113180
- Verification run: http://marvin5.arch.suse.cz/tests/728#

### CODE REVIEW

**Comment by grisu48** [comment]
Empty?

**Comment by jlausuch** [comment]
> Empty?

messed up the rebase... on it!

File: tests/containers/slem_rancher.pm [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    record_info('Kernel', 'Test that kernel files are present');
+    validate_script_output("podman exec slem_image /bin/sh -c 'ls /boot'", sub { /initrd/ });
+    validate_script_output("podman exec slem_image /bin/sh -c 'ls /boot'", sub { /vmlinuz/ });
+
+    record_info('Firmware', 'Test that /lib/firmware directory is not empty');
+    assert_script_run("podman exec slem_image /bin/sh -c 'test -d /lib/firmware'");
+    validate_script_output("podman exec slem_image /bin/sh -c 'rmdir /lib/firmware/' 2>&1", sub { /Directory not empty/ }, proceed_on_failure => 1);
+
+    record_info('grub', 'Test that /etc/default/grub exists');
+    assert_script_run("podman exec slem_image /bin/sh -c 'test -f /etc/default/grub'");
```
Comment (mloviska) [inline tests/containers/slem_rancher.pm:35]: Maybe `-s` might be better as it checks whether it is empty or not

File: tests/containers/slem_rancher.pm [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    record_info('repos', 'Image should come with empty repos');
+    validate_script_output("podman exec slem_image /bin/sh -c 'zypper lr' 2>&1", sub { /No repositories defined/ }, proceed_on_failure => 1);
+    assert_script_run("podman exec slem_image /bin/sh -c 'rmdir /etc/zypp/repos.d'");
```
Comment (mloviska) [inline tests/containers/slem_rancher.pm:39]: `ls -A` would return nothing, if the directory is empty. Maybe that might be more helpful when some repo would leak there accidentally.

**Comment by mloviska** [comment]
Minor comments, but good job. LGTM!

File: tests/containers/slem_rancher.pm (hunk 2 above)
Comment (jlausuch) [reply tests/containers/slem_rancher.pm:39]: yes, but then I'd need to use a comparison that `ls -A` doesn't return anything. rmdir will immediatelly fail if there is something in this dir.

File: tests/containers/slem_rancher.pm (hunk 1 above)
Comment (jlausuch) [reply tests/containers/slem_rancher.pm:35]: Much better, thanks for the suggestion!

File: tests/containers/slem_rancher.pm (hunk 2 above)
Comment (mloviska) [reply tests/containers/slem_rancher.pm:39]: yup, that would require another `validate_script_output`

File: tests/containers/slem_rancher.pm [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    assert_script_run("podman pull $image");
+    assert_script_run("podman run --name slem_image -dt $image");
+
+    record_info('Kernel', 'Test that kernel files are present');
+    validate_script_output("podman exec slem_image /bin/sh -c 'ls /boot'", sub { /initrd/ });
+    validate_script_output("podman exec slem_image /bin/sh -c 'ls /boot'", sub { /vmlinuz/ });
+
+    record_info('Firmware', 'Test that /lib/firmware directory is not empty');
+    assert_script_run("podman exec slem_image /bin/sh -c 'test -d /lib/firmware'");
+    validate_script_output("podman exec slem_image /bin/sh -c 'rmdir /lib/firmware/' 2>&1", sub { /Directory not empty/ }, proceed_on_failure => 1);
```
Comment (ilausuch) [inline tests/containers/slem_rancher.pm:32]: this is weird, could do that in a positive query? e.g.`[ -z "$(ls -A /lib/firmware)" ]`

File: tests/containers/slem_rancher.pm (hunk 2 above)
Comment (ilausuch) [reply tests/containers/slem_rancher.pm:39]: ah my previous comment was here commented

File: tests/containers/slem_rancher.pm (hunk 3 above)
Comment (jlausuch) [reply tests/containers/slem_rancher.pm:32]: good idea! thx

**Comment by jlausuch** [comment]
new VR: http://marvin5.arch.suse.cz/tests/771

File: tests/containers/slem_rancher.pm (hunk 2 above)
Comment (jlausuch) [reply tests/containers/slem_rancher.pm]: Please check now, following Ivan's suggestion.

Comment (mloviska) [reply tests/containers/slem_rancher.pm]: Me gusta! Buenisimo :)

**Review by pdostal — APPROVED** [review]

**Comment by jlausuch** [comment]
VR: http://marvin5.arch.suse.cz/tests/781

**Review by grisu48 — APPROVED** [review]

**Pull request merged**